### PR TITLE
Sqlite3/icu support

### LINKED ISSUE #9234: [package] sqlite3/3.*: Missing possibility to compile with the ICU extension (Unicode)
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **sqlite3/3.\***
  * Operating System+version: **All Platforms**
  * Conan version: **conan 1.44.1**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=Visual Studio
compiler.version=16
build_type=Release
[options]
[conf]
[build_requires]
[env]
```

### Additional Information

Hello! I would like to be able to use the [ICU extension](https://www.sqlite.org/src/dir?ci=trunk&name=ext/icu) for sqlite3 in a project. It's possible to enable it at compile time by defining the `SQLITE_ENABLE_ICU` variable ([source](https://sqlite.org/compile.html)). But then it needs to link to the `icu` package as well, which would add a dependency. 

I managed to got it partially working on a personal fork, here are the proposed changes:

* Add the `sqlite:enable_icu` option to the recipe, the same way the `enable_fs5` option works
* If this option is enabled, add a dependency to the icu package in the `requirements` method (but then what would be the right version to use?)

Should I open a pull request with the proposed change ?

### PR DESCRIPTION
**Pull request by kokounet**
Specify library name and version:  **sqlite3/\***

This fixes #9234 by adding the icu extension to the conan recipe.
It does so by adding the `enable_icu` option, defaulting to `False`.
When the `enable_icu` option is enabled,  the `sqlite3` package will depend on the `icu` package and define the `SQLITE_ENABLE_ICU` preprocessor variable.
I needed to add the `IMPORT_SUFFIX` part to `CMakeLists.txt:163` in order to fix a bug on Windows where the binary could not compile with ICU enabled due to a name clash.
The CXX language needs to be enabled on macOS in order to be able to compile.

I've tested on Windows and Mac and both seems to work.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally.

### CODE REVIEW

File: recipes/sqlite3/all/conanfile.py [hunk 1]
```diff
@@ -167,6 +170,10 @@ def _create_cmake_module_variables(module_file):
     def _module_file_rel_path(self):
         return os.path.join("lib", "cmake", "conan-official-{}-variables.cmake".format(self.name))
 
+    def requirements(self):
+        if self.options.enable_icu:
+            self.requires("icu/70.1")
+
```
Comment (SpaceIm) [inline recipes/sqlite3/all/conanfile.py:176]: Could you move this just before `validate()` please? It's easier to understand a recipe when methods are ordered by conan flow of execution.

**Comment by SpaceIm** [comment]
> The CXX language needs to be enabled on macOS in order to be able to compile.

icu is a C++ library with both a C & C++ API, so when icu is static, you need link the C++ standard library, this is why you had to enable C++ linker (I guess it would have also fail on Linux, not only macOS).

File: recipes/sqlite3/all/CMakeLists.txt [hunk 2]
```diff
@@ -127,6 +131,12 @@ if(THREADSAFE)
     target_link_libraries(${PROJECT_NAME}  PRIVATE Threads::Threads)
 endif()
 
+if(ENABLE_ICU)
+    find_package(ICU REQUIRED COMPONENTS data i18n uc)
```
Comment (SpaceIm) [inline recipes/sqlite3/all/CMakeLists.txt:135]: `cmake_find_package` generator should be added to `conanfile.py`

Comment (kokounet) [reply recipes/sqlite3/all/CMakeLists.txt:135]: I'm sorry, I don't get how I should change that part of the code. Should I add something to the `_module_file_rel_path` property or add it somewhere else?

Comment (SpaceIm) [reply recipes/sqlite3/all/CMakeLists.txt:135]: No, it's unrelated.
Just replace `generators = "cmake"` by `generators = "cmake", "cmake_find_package"`

**Pull request closed without merging**